### PR TITLE
[Web:Discover] Add more hints

### PR DESCRIPTION
**Pull request by kimlisa**
#### Description
Add more hints that points user to check their role permissions if resources aren't being detected such as here: https://github.com/gravitational/teleport/issues/21461

example screenshot for server:

image

### CODE REVIEW

File: web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx [hunk 1]
```diff
@@ -168,12 +168,17 @@ export function DownloadScript(
...
-        <Text mb={3}>
+        <Text mb={1}>
           - The Teleport Database Service could not join this Teleport cluster.
           Check the logs for errors by running{' '}
           <Mark>journalctl -fu teleport</Mark>.
         </Text>
 
+        <Text mb={3}>
+          - Check that your role permissions includes{' '}
+          <Mark>db_labels: '*': '*'</Mark>
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx:179]: wasn't sure if this was too restrictive or misinformation? b/c technically they can also define labels that matches their db resource too

Comment (zmb3) [reply web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx:179]: Does discover require that the user has access to _all_ resources?

Comment (kimlisa) [reply web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx:179]: > Does discover require that the user has access to all resources?

yes and no? if my understanding is correct, certain resources like `node`, we don't allow you to define your own labels yet, currently we deploy labelless servers, so asteriks are required. another way is to modify the resource with `tctl` afterwards and add the missing label (but that's probably more tedious vs using the UI to modify roles)

for other resources like kube and desktops (i only tested this with kube) technically you _can_ add/edit labels by modifying the output that we ask users to copy and paste but i don't think we expect users to modify these, so asteriks is also required here

for db,  I just tested that we don't allow a user to create a db resource if it doesn't satisfy the perms on their role eg: if they have `db_labels: env: '*'`, the new db resource also needs to define a label matching that criteria (using our label creator) else it gives me this error `"access to db denied. User does not have permissions. Confirm database user and name."` granted that's not a very clear message though... but after adding a `env: staging` label to my db resource, i successfully created a db resource.

so for dbs, i don't need to add this hint b/c the backend prevents it, but rest it looks like asteriks is required.

**Review by marcoandredinis — APPROVED** [review]

File: web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx (hunk 1 above)
Comment (ibeckermayer) [reply web/packages/teleport/src/Discover/Database/DownloadScript/DownloadScript.tsx]: > for db, I just tested that we don't allow a user to create a db resource if it doesn't satisfy the perms on their role eg: if they have db_labels: env: '*', the new db resource also needs to define a label matching that criteria (using our label creator) else it gives me this error "access to db denied. User does not have permissions. Confirm database user and name."

In that case, isn't this bullet point unnecessary? If the user can't create a resource that they don't have permission to access, then this won't be the reason they're not seeing it. Presuming that holds true across other resources (which it seems like it logically should, but may need to be checked), then we don't need it for the other resources either.

**Comment by kimlisa** [comment]
closing to revisit later

**Pull request closed without merging**